### PR TITLE
Fix #293441: [MusicXML export] invalid XML export: direction-type is …

### PR DESCRIPTION
**Pull request by lvinken**
…missing child element

Resolves: https://musescore.org/en/node/293441

TextBase::fragmentList() (last updated by commit 10ebbb9 on Aug 1, 2018) is missing a fix introduced in the similar function TextBase::plainText() (added by commit 3653f4b on Nov 19, 2018). The fix relayouts a clone of the TextBase when it encounters a TextBase with an invalid layout.

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [x] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
I can confirm that this fixes https://musescore.org/en/node/293441
Same issue and fix applies to 3.x too

File: src/importexport/musicxml/internal/musicxml/exportxml.cpp [hunk 1]
```diff
@@ -4336,6 +4336,7 @@ static void beatUnit(XmlWriter& xml, const TDuration dur)
 static void wordsMetronome(XmlWriter& xml, Score* s, TextBase const* const text, const int offset)
 {
     //LOGD("wordsMetronome('%s')", qPrintable(text->xmlText()));
```
Comment (RomanPudashkin) [inline src/importexport/musicxml/internal/musicxml/exportxml.cpp:4338]: Could you please remove these comments (and the one in MScoreTextToMXML::writeTextFragments)?

**Review by RomanPudashkin — APPROVED** [review]

File: src/importexport/musicxml/internal/musicxml/exportxml.cpp (hunk 1 above)
Comment (cbjeukendrup) [reply src/importexport/musicxml/internal/musicxml/exportxml.cpp:4338]: @lvinken Alternatively, if you would like to have some debug logging that's not activated by default but can be activated when you need to work on it, you can apply the same idea as in src/framework/audio/internal/audiobuffer.cpp:
```c++
//#define DEBUG_MUSICXML
#ifdef DEBUG_MUSICXML
#define LOG_MXML LOGD
#else
#define LOG_MXML LOGN
#endif

// and then:
LOG_MXML("wordsMetronome('%s')", qPrintable(text->xmlText()));

Comment (lvinken) [reply src/importexport/musicxml/internal/musicxml/exportxml.cpp]: @RomanPudashkin: done. @cbjeukendrup: thanks, good idea, unfortunately no time to implement that now.

**Comment by cbjeukendrup** [comment]
@lvinken A rebase is needed. It looks like the conflict was caused because `TextBase::_layout` was renamed to `TextBase::m_blocks`.

**Comment by lvinken** [comment]
Unable to do that right now, will have to wait a few weeks.Verstuurd vanaf mijn iPhoneOp 19 jun. 2023 om 12:56 heeft Casper Jeukendrup ***@***.***> het volgende geschreven:﻿
@lvinken A rebase is needed. It looks like the conflict was caused because TextBase::_layout was renamed to TextBase::m_blocks.

**Comment by Jojo-Schmitz** [comment]
I can help on that

**Comment by Jojo-Schmitz** [comment]
See #18056

**Comment by Jojo-Schmitz** [comment]
Came up again in https://musescore.org/en/node/351675

**Pull request closed without merging**